test/unit/notched-outline/index.test.tsx

```diff
@@ -100,57 +87,21 @@ test(
   () => {
     const wrapper = shallow<NotchedOutline>(<NotchedOutline />);
     wrapper.setProps({className: 'test-class-name'});
-    wrapper.instance().foundation_.notch = td.func();
-    wrapper.instance().foundation_.closeNotch = td.func();
+    wrapper.instance().foundation_.notch = td.func<(notchWidth: number) => void>();
+    wrapper.instance().foundation_.closeNotch = td.func<() => void>();
     td.verify(
       wrapper
         .instance()
-        .foundation_.notch(td.matchers.isA(Number), td.matchers.isA(Boolean)),
+        .foundation_.notch(td.matchers.isA(Number)),
       {times: 0}
     );
     td.verify(wrapper.instance().foundation_.closeNotch(), {times: 0});
   }
 );
 
-test('#adapter.getWidth returns width of outlineElement_', () => {
-  const div = document.createElement('div');
-  // needs to be attached to real DOM to get width
-  // https://github.com/airbnb/enzyme/issues/1525
-  document.body.append(div);
-  div.style.width = '150px';
-  div.style.height = '50px';
-  div.style.position = 'relative';
-  const options = {attachTo: div};
-  const wrapper = mount<NotchedOutline>(<NotchedOutline />, options);
-  const notchedOutlineElement = wrapper.find('.mdc-notched-outline').instance() as any;
-  const outlineWidth = notchedOutlineElement.offsetWidth;
-  assert.equal(
-    wrapper.instance().foundation_.adapter_.getWidth(),
-    outlineWidth
-  );
-  div.remove();
-});
-
-test('#adapter.getHeight returns height of outlineElement_', () => {
-  const div = document.createElement('div');
-  document.body.append(div);
-  div.style.width = '150px';
-  div.style.height = '50px';
-  div.style.position = 'relative';
-  const options = {attachTo: div};
-  const wrapper = mount<NotchedOutline>(<NotchedOutline />, options);
-  const notchedOutlineElement = wrapper.find('.mdc-notched-outline').instance() as any;
-  const outlineHeight = notchedOutlineElement.offsetHeight;
-  assert.equal(
-    wrapper.instance().foundation_.adapter_.getHeight(),
-    outlineHeight
-  );
-  div.remove();
-});
-
 test('#adapter.addClass adds class to classList', () => {
   const wrapper = shallow<NotchedOutline>(<NotchedOutline />);
-  wrapper.instance().foundation_.adapter_.addClass('test-class-name');
+  getAdapter(wrapper.instance().foundation_).addClass('test-class-name');
```

Comment:
Is this method necessary?

Reply:
it allows to put code with ts-ignore in one place. The adapter_ property is protected in mdc-web